flink-table/flink-table-common/src/main/java/org/apache/flink/table/utils/PrintUtils.java

```diff
@@ -116,65 +116,60 @@ public static void printAsTableauForm(
             String nullColumn,
             boolean deriveColumnWidthByType,
             boolean printRowKind) {
-        final List<TableColumn> columns = tableSchema.getTableColumns();
-        String[] columnNames = columns.stream().map(TableColumn::getName).toArray(String[]::new);
-        if (printRowKind) {
-            columnNames =
-                    Stream.concat(Stream.of(ROW_KIND_COLUMN), Arrays.stream(columnNames))
-                            .toArray(String[]::new);
-        }
-
-        final int[] colWidths;
-        if (deriveColumnWidthByType) {
-            colWidths =
-                    columnWidthsByType(
-                            columns,
-                            maxColumnWidth,
-                            nullColumn,
-                            printRowKind ? ROW_KIND_COLUMN : null);
-        } else {
-            final List<Row> rows = new ArrayList<>();
-            final List<String[]> content = new ArrayList<>();
-            content.add(columnNames);
-            while (it.hasNext()) {
-                Row row = it.next();
-                rows.add(row);
-                content.add(rowToString(row, nullColumn, printRowKind));
+        if (it.hasNext()) {
```

Comment:
Personally, I don't like big if else block. Could you return the method when it is emtpy?